### PR TITLE
Version 2.11 - Mark secrets file as ghost file in rpm spec

### PR DESCRIPTION
**Pull request by felixsch**
## Description

card: https://trello.com/c/JbsuXdpH/2726-rmt-downgrade-210-29-broken

To allow downgrades without file permission errors created due to a missing managed file ('/usr/share/rmt/config/secrets.yml.key) it needs to be tracked as `ghost` file in the rpm spec file.

**How to test this pull request:**

- build the rpm for rmt-server-2.11
```
rmt> make dist
rmt> cp -r obs/package/* <path/to/your/rpm> (or osc co systemsmanagement:SCC:RMT rmt-server)
rpm> build --no-verify --clean SLE_15_SP3 (or what every sp you have docker around)
```

- install and downgrade packages
```
cd /var/tmp/build-root/SLE_15_<YOUR SERVICE PACK!>-x86_64/home/abuild/rpmbuild/RPMS/x86_64

docker run -v (pwd):/rmt --privileged --rm --network=host -h rmt-review -ti connect.15sp3 /bin/bash

> zypper in rmt-server-config-2.11-0.x86_64.rpm rmt-server-2.11-0.x86_64.rpm
> SUSEConnect -r <a regcode of your choice>
> zypper in --oldpackage rmt-server-2.10

#! Check that the installation succeeds without breaking! (aka. permission issues)

```

### CODE REVIEW

File: package/obs/rmt-server.spec [hunk 1]
```diff
@@ -251,6 +251,10 @@ chrpath -d %{buildroot}%{lib_dir}/vendor/bundle/ruby/*/extensions/*/*/mysql2-*/m
 %dir /var/lib/rmt
 %ghost %{_datadir}/rmt/public/repo
 %ghost %{_datadir}/rmt/public/suma
+
+# The secrets file is created by running the initial rake tasks in the `post` section
+%ghost %{app_dir}/config/secrets.yml.key
```
Comment (ngetahun) [inline package/obs/rmt-server.spec:256]: I think you forgot the `secrets.yml.enc` file too

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/300947/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 5b11f2e4dfb42365f84cfa839e9eddd6d597023c into ecf699d0d093ee80b60035422c1d773abed5adc3)

File: package/obs/rmt-server.spec (hunk 1 above)
Comment (felixsch) [reply package/obs/rmt-server.spec:256]: absolutely. I even missed copying the right file into the project! I added it.

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/300969/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 0321b6c5252cd060fabf4d46d1a6608188f9a950 into ecf699d0d093ee80b60035422c1d773abed5adc3)

**Review by ngetahun — APPROVED** [review]
LGTM!

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/302243/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged e082ab86f418a8543a798055a5516abeb20c76a5 into c5cedbbb50bfadc7f9ec1c224c0f6278c6fa3e46)

**Pull request merged**